generate/templates/templates/nodegit.js

```diff
@@ -1,4 +1,30 @@
-var promisify = require("promisify-node");
+var promisifyNode = require("promisify-node");
+
+var asyncPromise = Promise.resolve();
+
+function promisify(fn) {
+  function chainedPromisified() {
+    var promisified = promisifyNode(fn);
```

Comment:
This should be up in the previous scope so we don't promisify the function every time it's called.